web_programming/fetch_anime_and_play.py

```diff
@@ -0,0 +1,203 @@
+        anime_list = []
+        for li in items_li:
+            if not isinstance(li, NavigableString):
+                anime_url, anime_title = li.find("a")["href"], li.find("a")["title"]
+                anime_list.append(
+                    {
+                        "title": anime_title,
+                        "url": anime_url,
+                    }
+                )
+
+        return anime_list
+
+    except (requests.exceptions.RequestException, HTTPError, TypeError) as e:
+        raise e
+
+
+def search_anime_episode_list(episode_endpoint: str) -> list:
+
+    """[summary]
+
+    This function will take an url and
+    return list of episodes after scraping the site
+    for an url.
+
+    >>> type(search_anime_episode_list("/anime/kimetsu-no-yaiba"))
+    <class 'list'>
+
+    Args:
+        episode_endpoint (str): [Endpoint of episode]
+
+    Raises:
+        e: [description]
+
+    Returns:
+        [list]: [List of episodes]
+    """
+
+    try:
+
+        request_url = f"{BASE_URL}{episode_endpoint}"
+        response = requests.get(
+            url=request_url, headers={"UserAgent": UserAgent().chrome}
+        )
+        soup = BeautifulSoup(response.text, "html.parser")
+
+        # With this id. get the episode list.
+        episode_page_ul = soup.find("ul", {"id": "episode_related"})
+        episode_page_li = episode_page_ul.children
+
+        episode_list = []
+        for children in episode_page_li:
+            try:
+                if not isinstance(children, NavigableString):
+                    episode_list.append(
+                        {
+                            "title": children.find(
+                                "div", {"class": "name"}
+                            ).text.replace(" ", ""),
+                            "url": children.find("a")["href"],
+                        }
+                    )
+            except (KeyError, NotFoundErr, TypeError):
+                pass
+
+        return episode_list
+
+    except (requests.exceptions.RequestException) as e:
+        raise e
+
+
+def get_anime_episode(episode_endpoint: str) -> list:
+
+    """[summary]
+
+    Get click url and download url from episode url
+
+    Args:
+        episode_endpoint (str): [Endpoint of episode]
+
+    Raises:
+        e: [description]
+
+    Returns:
+        [list]: [List of download and watch url]
+    """
+
+    try:
+
+        episode_page_url = f"{BASE_URL}{episode_endpoint}"
+
+        response = requests.get(
+            url=episode_page_url, headers={"User-Agent": UserAgent().chrome}
+        )
+        soup = BeautifulSoup(response.text, "lxml")
+
+        episode_url = soup.find("iframe", {"id": "playerframe"})["src"]
+        download_url = episode_url.replace("/embed/", "/playlist/") + ".m3u8"
+        return [f"{BASE_URL}{episode_url}", f"{BASE_URL}{download_url}"]
+
+    except (
+        KeyError,
+        NotFoundErr,
+        TypeError,
+        requests.exceptions.RequestException,
```

Comment:
Long try / except blocks lead to many different exceptions which confuses the reader of the code.  It also slows down debugging when one of them is raised.

Reply:
removed multiple exceptions